File: exercise/python/day06/workshop/elf_workshop.py
class ElfWorkshop:
    def __init__(self):
        self.task_list = []

    def get_task_list(self):
        return self.task_list

    def add_task(self, task):
        if task:
            self.task_list.append(task)

    def complete_task(self):
        if self.task_list:
            return self.task_list.pop(0)
        return None

File: exercise/scala/day06/src/main/scala/workshop/ElfWorkshop.scala
package workshop

import scala.collection.mutable.ListBuffer

class ElfWorkshop {
  var taskList: ListBuffer[String] = ListBuffer()

  def getTaskList: List[String] = taskList.toList

  def addTask(task: String): Unit = {
    if (Option(task).exists(_.nonEmpty)) {
      taskList += task
    }
  }

  def completeTask(): String = if (taskList.nonEmpty) taskList.remove(0) else null
}